### PR TITLE
feat(volume share): add share volume REST API

### PR DESCRIPTION
**Pull request by paulyoong**
Add the ability to share a volume to the REST API.

Modify the share API for all resources (nexuses and replicas) such that
only the appropriate protocols show up. For example, a replica will only
be permitted to be shared over nvmf.

Resolves: CAS-798

### CODE REVIEW

File: control-plane/mbus-api/src/v0.rs [hunk 1]
```diff
@@ -665,6 +665,83 @@ impl From<i32> for Protocol {
...
+    fn default() -> Self {
+        Self::Nvmf
+    }
+}
+impl From<i32> for NexusShareProtocol {
+    fn from(src: i32) -> Self {
+        match src {
+            1 => Self::Nvmf,
+            2 => Self::Iscsi,
+            _ => panic!("Invalid"),
```
Comment (tiagolobocastro) [inline control-plane/mbus-api/src/v0.rs:707]: maybe display what the invalid src was?
same for below

File: control-plane/rest/openapi-specs/v0_api_spec.json [hunk 2]
```diff
@@ -1 +1 @@
...
...$ref":"#/definitions/RestJsonError"}},"408":{"description":"Bad Request","schema":{"$ref":"#/definitions/RestJsonError"}},"412":{"description":"Precondition Failed","schema":{"$ref":"#/definitions/RestJsonError"}},"416":{"description":"Range Not satisfiable","schema":{"$ref":"#/definitions/RestJsonError"}},"422":{"description":"Unprocessable entity","schema":{"$ref":"#/definitions/RestJsonError"}},"500":{"description":"Internal Server Error","schema":{"$ref":"#/definitions/RestJsonError"}},"501":{"description":"Not Implemented","schema":{"$ref":"#/definitions/RestJsonError"}},"503":{"description":"Service Unavailable","schema":{"$ref":"#/definitions/RestJsonError"}},"504":{"description":"Gateway Timeout","schema":{"$ref":"#/definitions/RestJsonError"}},"507":{"description":"Insufficient Storage","schema":{"$ref":"#/definitions/RestJsonError"}}},"parameters":[{"description":"UUID of a mayastor volume","in":"path","name":"volume_id","required":true,"type":"string"}],"tags":["Volumes"]}}},"basePath":"/v0","securityDefinitions":{"JWT":{"name":"Authorization","type":"apiKey","in":"header","description":"Use format 'Bearer TOKEN'"}},"info":{"version":"v0","title":"Mayastor RESTful API"}}
```
Comment (tiagolobocastro) [inline control-plane/rest/openapi-specs/v0_api_spec.json:1]: As an aside, I've been wondering, should we use a yaml format instead? more readable :)

**Review by tiagolobocastro — APPROVED** [review]

File: control-plane/mbus-api/src/v0.rs (hunk 1 above)
Comment (paulyoong) [reply control-plane/mbus-api/src/v0.rs]: Good suggestion. Done.

File: control-plane/rest/openapi-specs/v0_api_spec.json (hunk 2 above)
Comment (paulyoong) [reply control-plane/rest/openapi-specs/v0_api_spec.json:1]: I think that is a good idea :+1:

**Comment by paulyoong** [comment]
bors merge

**Comment by paulyoong** [comment]
bors merge

**Pull request merged**